src/main/java/corgi/Corgi.java

```diff
@@ -0,0 +1,84 @@
+package corgi;
+import java.util.Scanner;
+
+import corgi.commands.Command;
+import corgi.commands.CommandExecutionException;
+import corgi.parsers.CommandParser;
+import corgi.parsers.InvalidCommandFormatException;
+import corgi.parsers.InvalidCommandTypeException;
+import corgi.parsers.TaskParser;
+import corgi.storage.Storage;
+import corgi.tasks.Task;
+import corgi.tasks.TaskList;
+import corgi.ui.Ui;
+
+/**
+ * The `Corgi` class represents a chatbot named Corgi that manages tasks.
+ * It provides a command-line interface for interacting with the chatbot.
+ *
+ * This class initializes the chatbot and handles user input and commands.
+ */
+public class Corgi {
+    private TaskList tasks;
+    private Storage<Task> storage;
+    private Ui ui;
+
+    /**
+     * Constructs new Corgi chatbot with an empty task list.
+     */
+    public Corgi() {
+        this.ui = new Ui();
+        this.storage = new Storage<>(new TaskParser(), "./data/tasks.txt");
+        this.tasks = new TaskList(storage.load());
+
+        if (tasks.size() > 0) {
+            this.ui.showTasksLoaded(tasks.size());
+        }
+    }
+
+    /**
+     * Starts the chatbot - Corgi.
+     */
+    public void start() {
+        Scanner sc = new Scanner(System.in);
```

Comment:
Would it be better to give the scanner a more descriptive name, given that it has a scope of > 20 lines?